WordPress/Classes/ViewRelated/Blog/SharingDetailViewController.m

```diff
@@ -180,7 +174,37 @@ - (void)reconnectPublicizeConnection
         return;
     }
 
-    [self.helper reconnectPublicizeConnection:self.publicizeConnection];
+    SharingService *sharingService = [[SharingService alloc] initWithManagedObjectContext:[self managedObjectContext]];
+    self.helper = [self authorizationHelper];
+
+    __weak __typeof(self) weakSelf = self;
+    void (^attemptReconnect)() = ^{
+        weakSelf.helper = [self authorizationHelper];
+        [weakSelf.helper reconnectPublicizeConnection:weakSelf.publicizeConnection];
+    };
+    if (self.helper == nil) {
+        [sharingService syncPublicizeServicesForBlog:self.blog
+                                             success:attemptReconnect
+                                             failure:^(NSError * _Nullable error) {
+                                                 [WPError showNetworkingAlertWithError:error];
+                                             }];
+    } else {
+        attemptReconnect();
+    }
+}
+
+- (SharingAuthorizationHelper *)authorizationHelper
```

Comment:
The only reason we keep the property is so we can set the delegate to nil on dealloc.